### PR TITLE
adding selectLookup method under VBlock

### PR DESCRIPTION
**Pull request by h-haddad**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt [hunk 1]
```diff
@@ -2469,6 +2553,41 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
...
+
+    for (field in fields) {
+
+      val col: Column<Any>? = if (field.isNull(recno) || !field.hasIndex(idx)) {
+        null
+      } else {
+        field.lookupColumn_(tables!![0])
+      }
+      if (col != null) {
+        buffer.add(Op.build { col eq field.getSql(recno)!! })
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt:2570]: we dont need Op.build

File: galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt [hunk 2]
```diff
@@ -2469,6 +2553,41 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
...
+
+        if (query.execute(this)!!.next()) {
+            if (query.first()[idColumn_] != id) {
+              form.setActiveBlock(this@VBlock)
+              activeRecord = recno
+              gotoFirstField()
+              throw VExecFailedException(MessageCode.getMessage("VIS-00014", arrayOf<Any>(indices!![idx])))
+            }
+          assert(!query.execute(this)!!.next()) { "too many rows" }
+        }
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt:2586]: you can optimize here by using .single() that will retrieve one element from list or throw exceptions instead

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 3]
```diff
@@ -18,6 +18,8 @@
 
 package org.kopi.galite.form
 
+import org.jetbrains.exposed.sql.Column
+import org.jetbrains.exposed.sql.Table
 import org.jetbrains.exposed.sql.transactions.transaction
 import java.awt.Color
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:24]: Fix imports order

**Review by ZMarouani — CHANGES_REQUESTED** [review]

**Review by ZMarouani — APPROVED** [review]

**Review by SNsir — CHANGES_REQUESTED** [review]

**Review by SNsir — APPROVED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/type/Timestamp.kt [hunk 4]
```diff
@@ -18,14 +18,30 @@
...
 
   fun compareTo(other: Timestamp): Int = TODO()
 
   fun add(millis: Long): NotNullTimestamp = TODO()
 
+  /**
+   * Represents the value in sql
+   */
+  open fun toSql(): String? {
+    val tmp = StringBuffer()
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/type/Timestamp.kt:33]: use `buildString`

**Pull request merged**